gems/aws-sdk-s3/lib/aws-sdk-s3/plugins/s3_signer.rb

```diff
@@ -175,7 +175,7 @@ def custom_endpoint?(resp)
           end
 
           def wrong_sigv4_region?(resp)
-            resp.context.http_response.status_code == 400 &&
+            [400, 301].include?(resp.context.http_response.status_code) &&
```

Comment:
Is this intended for this change? (I typically agree we should be doing this... but maybe not as part of this)

Reply:
Good catch - removed from this PR.